IO/GeoFileParser.cpp

```diff
@@ -715,61 +715,130 @@ Goal* GeoFileParser::parseGoalNode(TiXmlElement * e)
+               wa->SetSubRoomID(value);
+          } else {
+               Log->Write("ERROR:\t  waiting area %d: subroom_id set but not an integer", wa->GetId());
+               delete wa;
+               return nullptr;
+          }
+     } else {
+          Log->Write("ERROR:\t  waiting area %d: subroom_id required", wa->GetId());
+          delete wa;
           return nullptr;
      }
 
-     if ((min_peds < 0 || max_peds < 0 || waiting_time < 0) && transition_id < 0){
-          Log->Write("ERROR:\t  WA  %d: min_peds, max_peds, transition_id not set properly. "
-                     "Set either (min_peds, max_peds, waiting_time) or transition_id.", id);
-          return nullptr;
+     // Read caption and check if correct value
+     if(const char* attribute = e->Attribute("caption"); attribute) {
+          if (std::string value = xmltoa(attribute, "-1"); !value.empty()) {
+               wa->SetCaption(value);
+          }
      }
 
+     Log->Write("INFO:\t  Goal id: %d", wa->GetId());
+     Log->Write("INFO:\t  Goal caption: %s", wa->GetCaption().c_str());
+     Log->Write("INFO:\t  Goal room_id: %d", wa->GetRoomID());
+     Log->Write("INFO:\t  Goal subroom_id: %d", wa->GetSubRoomID());
 
-     std::string caption = xmltoa(e->Attribute("caption"), "-1");
+     // Read optional values and check for valid values, on fail write error
+     // Read min_peds and check if correct value
+     if(const char* attribute = e->Attribute("min_peds"); attribute) {
+          if (int value = xmltoi(attribute, -1); value>0) {
+               wa->SetMinNumPed(value);
+          }
+     }
 
-     Log->Write("INFO:\t  Goal id: %d", id);
-     Log->Write("INFO:\t  Goal caption: %s", caption.c_str());
-     Log->Write("INFO:\t  Goal room_id: %d", room_id);
-     Log->Write("INFO:\t  Goal subroom_id: %d", subroom_id);
+     // Read max_peds and check if correct value
+     if(const char* attribute = e->Attribute("max_peds"); attribute) {
+          if (int value = xmltoi(attribute, -1); value>0) {
+               wa->SetMaxNumPed(value);
+          }
+     }
+
+     // Read waiting_time and check if correct value
+     if(const char* attribute = e->Attribute("waiting_time"); attribute) {
+          if (double value = xmltof(attribute, -1.); value>=0.) {
+               wa->SetWaitingTime(value);
+          }
+     }
+
+     // Read transition_id and check if correct value
+     if(const char* attribute = e->Attribute("transition_id"); attribute) {
+          if (int value = xmltoi(attribute, -1); value>-1) {
+               wa->SetTransitionID(value);
+          }
+     }
+
+     // Read is_open and check if correct value
+     if(const char* attribute = e->Attribute("is_open"); attribute) {
+          if (bool value = strcmp(xmltoa(attribute, "false"), "true") == 0) {
+               wa->SetOpen(value);
+          }
+     }
+
+     // Read global_timer and check if correct value
+     if(const char* attribute = e->Attribute("global_timer"); attribute) {
```

Comment:
Same as for is_open. Also this attribute is not mentioned in the XSD, @chraibi is the XSD still updated?